ecs-cli/modules/config/config_v1.go

```diff
@@ -84,73 +84,84 @@ func NewCLIConfig(cluster string) *CLIConfig {
 // ToAWSSession creates a new Session object from the CliConfig object.
 //
 // Region: Order of resolution
-//  1) Environment Variable - attempts to fetch the region from environment variables:
+//  1) ECS CLI Flags
+//   a) Region Flag --region
+//   b) Cluster Config Flag (--cluster-config)
+//  2) ECS Config - attempts to fetch the region from the default ECS Profile
+//  3) Environment Variable - attempts to fetch the region from environment variables:
 //    a) AWS_REGION (OR)
 //    b) AWS_DEFAULT_REGION
-//  2) ECS Config - attempts to fetch the region from the ECS config file
-//  3) AWS Profile - attempts to use region from AWS profile name
-//    a) profile name from ECS config file (OR)
+//  4) AWS Profile - attempts to use region from AWS profile name
 //    b) AWS_PROFILE environment variable (OR)
 //    c) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
 //
 // Credentials: Order of resolution
-//  1) Environment Variable - attempts to fetch the credentials from environment variables:
-//   a) AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY (OR)
-//   b) AWS_ACCESS_KEY and AWS_SECRET_KEY
-//  2) ECS Config - attempts to fetch the credentials from the ECS config file
-//  3) AWS Profile - attempts to use credentials (aws_access_key_id, aws_secret_access_key) or assume_role (role_arn, source_profile) from AWS profile name
-//    a) profile name from ECS config file (OR)
-//    b) AWS_PROFILE environment variable (OR)
-//    c) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
-//  4) EC2 Instance role
-func (cfg *CLIConfig) ToAWSSession() (*session.Session, error) {
-	svcConfig := aws.Config{}
-	return cfg.toAWSSessionWithConfig(svcConfig)
+//  1) ECS CLI Profile Flags
+//   a) ECS Profile (--ecs-profile)
+//   b) AWS Profile (--aws-profile)
+//  2) Environment Variables - attempts to fetch the credentials from environment variables:
+//   a) ECS_PROFILE
+//   b) AWS_PROFILE
+//   c) AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY, Optional: AWS_SESSION_TOKEN
+//  3) ECS Config - attempts to fetch the credentials from the default ECS Profile
+//  4) Default AWS Profile - attempts to use credentials (aws_access_key_id, aws_secret_access_key) or assume_role (role_arn, source_profile) from AWS profile name
+//    a) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
+//  5) EC2 Instance role
+func (cfg *CLIConfig) ToAWSSession(context *cli.Context) (*session.Session, error) {
+
+	region := cfg.getRegion()
+
+	if isProfileFlagsCase(context) {
+		return credsFromECSConfig(cfg)
+	} else if isEnvVarCase(context) {
+		return defaultProvider(region)
+	} else if isDefaultECSProfileCase(cfg) {
+		return credsFromECSConfig(cfg)
+	} else {
+		return defaultProvider(region)
+	}
```

Comment:
Based on the naming inconsistencies, I wonder if it might be better to refactor this so that you instantiate a Session object at the start of this function, and then set the proper credentials on that Session object. based on methods named something like `defaultCreds` or `ecsProfileCreds` etc.

Reply:
So in the updated code I had to add a new function `defaultProviderFromProfile()`. This new function instantiates the session in a different way than all the others, and changing it to work like you suggested causes the test cases to fail :( 

I definitely agree that this code is a bit repetitive though.